core/src/avm2/activation.rs

```diff
@@ -0,0 +1,1627 @@
+        context: &mut UpdateContext<'_, 'gc, '_>,
+    ) -> Result<FrameControl<'gc>, Error> {
+        //TODO: After adding ints, change this to ints.
+        let cur_index = self.avm2.pop().as_number()?;
+        let mut object = self.avm2.pop().as_object()?;
+
+        let name = object.get_enumerant_name(cur_index as u32);
+        let value = if let Some(name) = name {
+            object.get_property(object, &name, self, context)?
+        } else {
+            Value::Undefined
+        };
+
+        self.avm2.push(value);
+
+        Ok(FrameControl::Continue)
+    }
+
+    fn op_is_type(
+        &mut self,
+        method: Gc<'gc, BytecodeMethod<'gc>>,
+        context: &mut UpdateContext<'_, 'gc, '_>,
+        type_name_index: Index<AbcMultiname>,
+    ) -> Result<FrameControl<'gc>, Error> {
+        let value = self.avm2.pop().as_object()?;
+
+        let type_name = self.pool_multiname_static(method, type_name_index)?;
+        let type_object = if let Some(scope) = self.scope() {
+            scope.read().find(&type_name, self, context)?
+        } else {
+            None
+        };
+
+        if let Some(type_object) = type_object {
+            let is_instance_of = value.is_instance_of(self, context, type_object, true)?;
+            self.avm2.push(is_instance_of);
+        } else {
+            return Err(format!(
+                "Attempted to check against nonexistent type {:?}",
+                type_name
+            )
+            .into());
+        }
+
+        Ok(FrameControl::Continue)
+    }
+
+    fn op_is_type_late(
+        &mut self,
+        context: &mut UpdateContext<'_, 'gc, '_>,
+    ) -> Result<FrameControl<'gc>, Error> {
+        let type_object = self.avm2.pop().as_object()?;
+        let value = self.avm2.pop().as_object()?;
+
+        let is_instance_of = value.is_instance_of(self, context, type_object, true)?;
+
+        self.avm2.push(is_instance_of);
+
+        Ok(FrameControl::Continue)
+    }
+
+    fn op_instance_of(
+        &mut self,
+        context: &mut UpdateContext<'_, 'gc, '_>,
+    ) -> Result<FrameControl<'gc>, Error> {
+        let type_object = self.avm2.pop().as_object()?;
+        let value = self.avm2.pop().as_object()?;
+
+        let is_instance_of = value.is_instance_of(self, context, type_object, false)?;
+
+        self.avm2.push(is_instance_of);
+
+        Ok(FrameControl::Continue)
+    }
+
+    #[allow(unused_variables)]
+    fn op_debug(
+        &mut self,
+        method: Gc<'gc, BytecodeMethod<'gc>>,
+        is_local_register: bool,
+        register_name: Index<String>,
+        register: u8,
+    ) -> Result<FrameControl<'gc>, Error> {
+        if is_local_register {
+            let register_name = self.pool_string(method, register_name)?;
+            let value = self.local_register(register as u32)?;
+
+            avm_debug!("Debug: {} = {:?}", register_name, value);
+        } else {
+            avm_debug!("Unknown debugging mode!");
+        }
```

Comment:
We can wrap this whole thing behind `avm_debug` feature flag, and save the lookups.